tools/swift-stdlib-tool/CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 add_swift_host_tool(swift-stdlib-tool
-  swift-stdlib-tool.mm
+  swift-stdlib-tool.cpp
   SWIFT_COMPONENT compiler)
 
-find_library(FOUNDATION NAMES Foundation)
-target_link_libraries(swift-stdlib-tool PRIVATE ${FOUNDATION})
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
This implicitly means the entire project now requires a C++17-enabled compiler. Please mention this change on swift evolution.

Reply:
To be clear, I'm not opposed to such a change, but we have a wide variety of adopters with a wide variety of toolchains and we must be careful to make sure we're advertising the correct minimum requirements.

Reply:
Do all of those toolchains need this tool, or is it limited to certain platforms like Darwin?

Reply:
Right now, this tool is built to ship specifically for macOS hosts in Darwin toolchains.

Reply:
That should dramatically reduce impact, because there is only one variety of toolchain we need to worry about compatibility with, then, right?

Reply:
I can mention this in the forums. Xcode 13 beta 4 is [currently required](https://github.com/apple/swift/blob/main/docs/HowToGuides/GettingStarted.md#installing-dependencies) on macOS and includes a C++17 enabled version of Clang so I thought it would be ok to use C++17 here.

Reply:
> That should dramatically reduce impact, because there is only one variety of toolchain we need to worry about compatibility with, then, right?

Yes.

> so I thought it would be ok to use C++17 here.

It is okay to use. I just want our adopters to be aware of this change, that's all.

Reply:
Is there a reason to use `CMAKE_CXX_STANDARD` rather than setting [`CXX_STANDARD`](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html#prop_tgt:CXX_STANDARD) on just this target? I know you don't want to diverge too much across the project, but it seems like if you don't have a C++17-capable stdlib, you could just skip building this one target.

Reply:
Ah, I'm not very familiar with CMake and didn't know this was possible. I'll use CXX_STANDARD on just this target instead. Thank you.